### ISSUE TITLE
Show all available cores + Support core selection

### ISSUE THREAD

**Issue by GoogleCodeExporter** [JPaulMora/Pyrit#237]
```
What steps will reproduce the problem?
1. Build and install pyrit.
2. list_cores / benchmark.
3. Build and install cpyrit_opencl.
4. list_cores / benchmark.
5. Uninstall cpyrit_opencl.
6. Build and install cpyrit_cuda.
7. list_cores / benchmark.
8. Install cpyrit_opencl.
9. list_cores / benchmark.

What is the expected output? What do you see instead?
I should see all available cores that are supported by installed libs and they 
should be listed only once. But I get following output (relevant parts):

2.
#1:  'CPU-Core (SSE2)'
#2:  'CPU-Core (SSE2)'
#3:  'Network-Clients'

4.
Running benchmark (13788.5 PMKs/s)... \ 

Computed 13788.48 PMKs/s total.
#1: 'OpenCL-Device 'GeForce GTX 460'': 14340.2 PMKs/s (RTT 2.7)
#2: 'CPU-Core (SSE2)': 277.1 PMKs/s (RTT 3.0)
#3: 'Network-Clients': 0.0 PMKs/s (RTT 0.0)

7.
Running benchmark (14339.5 PMKs/s)... \ 

Computed 14339.46 PMKs/s total.
#1: 'CUDA-Device #1 'GeForce GTX 460'': 15166.8 PMKs/s (RTT 2.8)
#2: 'CPU-Core (SSE2)': 232.8 PMKs/s (RTT 3.1)
#3: 'Network-Clients': 0.0 PMKs/s (RTT 0.0)

9.
Running benchmark (14270.4 PMKs/s)... - 

Computed 14270.42 PMKs/s total.
#1: 'CUDA-Device #1 'GeForce GTX 460'': 8421.5 PMKs/s (RTT 2.7)
#2: 'OpenCL-Device 'GeForce GTX 460'': 6798.7 PMKs/s (RTT 2.8)
#3: 'Network-Clients': 0.0 PMKs/s (RTT 0.0)

What version of the product are you using? On what operating system?
Pyrit 0.4.0-dev (svn r288)
Debian squeeze (6.0) x86_64
nvidia+cuda+opencl stuff from Debian unstable (260.19.21-1, 3.2.16-1)
Athlon 64 X2 4600+
GeForce GTX 460

Please provide any additional information below.
There should be an option (e.g. called -c) for selecting which cores should be 
used. Omitting it would mean that all available cores will be used (obviously 
it's not smart if we have support for opencl and cuda together installed). 
Benchmark should actually benchmark each (mentioned) core alone and only in the 
end try combined power of all cores.
```

Original issue reported on code.google.com by `prze...@gmail.com` on 14 Jan 2011 at 9:29
- Merged into: #201
